tests/test_complete_run.py

```diff
@@ -0,0 +1,111 @@
+        ts_param.test_name = 'e3sm_v1'
+        ts_param.start_yr = '1990'
+        ts_param.end_yr = '1999'
+
+        runner.sets_to_run = ['lat_lon','zonal_mean_xy', 'zonal_mean_2d', 'polar', 'cosp_histogram', 'meridional_mean_2d', 'enso_diags', 'qbo','area_mean_time_series']
+        runner.run_diags([param, enso_param, qbo_param, ts_param])
+
+        actual_images_dir = './tutorial_2020_all_sets'
+        # The expected_images_file lists all 475 images we expect to compare.
+        # It was generated with the following steps:
+        # cd /lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/tutorial_2020_all_sets
+        # find . -type f -name '*.png' > expected_images_complete_run.txt
+        # Move file to e3sm_diags repository
+        expected_images_file = 'expected_images_complete_run.txt'
+        expected_images_dir = '/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/tutorial_2020_all_sets'
+
+        mismatched_images = []
+
+        with open(expected_images_file) as f:
+            for line in f:
+                image = line.strip('./').strip('\n')
+                path_to_actual_png = os.path.join(actual_images_dir, image)
+                path_to_expected_png = os.path.join(expected_images_dir, image)
+
+                actual_png = Image.open(path_to_actual_png).convert('RGB')
+                expected_png = Image.open(path_to_expected_png).convert('RGB')
+                diff = ImageChops.difference(actual_png, expected_png)
+
+                bbox = diff.getbbox()
+                if not bbox:
+                    # If `diff.getbbox()` is None, then the images are in theory equal
+                    self.assertIsNone(diff.getbbox())
+                else:
+                    # Sometimes, a few pixels will differ, but the two images appear identical.
+                    # https://codereview.stackexchange.com/questions/55902/fastest-way-to-count-non-zero-pixels-using-python-and-pillow
+                    nonzero_pixels = diff.crop(bbox).point(lambda x: 255 if x else 0).convert("L").point(bool).getdata()
+                    num_nonzero_pixels = sum(nonzero_pixels)
+                    print('\npath_to_actual_png={}'.format(path_to_actual_png))
+                    print('path_to_expected_png={}'.format(path_to_expected_png))
+                    print('diff has {} nonzero pixels.'.format(num_nonzero_pixels))
+                    width, height = expected_png.size
+                    num_pixels = width * height
+                    print('total number of pixels={}'.format(num_pixels))
+                    fraction = num_nonzero_pixels/num_pixels
+                    print('num_nonzero_pixels/num_pixels fraction={}'.format(fraction))
+                    # Fraction of mismatched pixels should be less than 0.01%
+                    if fraction >= 0.0001:
+                        mismatched_images.append(image)
+
+        self.assertEqual(mismatched_images, [])
```

Comment:
The test will check all images before returning pass/fail, rather than simply failing on the first mismatched image.